src/nrniv/CMakeLists.txt

```diff
@@ -304,11 +304,20 @@ if(NRN_USE_BACKWARD)
     PROPERTY COMPILE_DEFINITIONS USE_BACKWARD=1)
 endif()
 
-# NVHPC/21.7 cannot compile znorm.c with -O2 or above. See also:
-# https://forums.developer.nvidia.com/t/nvc-21-7-regression-internal-compiler-error-can-only-coerce-indirect-args/184847
-if(NRN_HAVE_NVHPC_COMPILER AND ${CMAKE_C_COMPILER_VERSION} VERSION_EQUAL 21.7)
-  set_source_files_properties(${PROJECT_SOURCE_DIR}/src/mesch/znorm.c PROPERTIES COMPILE_OPTIONS
-                                                                                 -Mnovect)
+if(NRN_HAVE_NVHPC_COMPILER)
+  # NVHPC/21.7 cannot compile znorm.c with -O2 or above. See also:
+  # https://forums.developer.nvidia.com/t/nvc-21-7-regression-internal-compiler-error-can-only-coerce-indirect-args/184847
+  if(${CMAKE_C_COMPILER_VERSION} VERSION_EQUAL 21.7)
+    set_source_files_properties(${PROJECT_SOURCE_DIR}/src/mesch/znorm.c PROPERTIES COMPILE_OPTIONS
+                                                                                   -Mnovect)
+  endif()
+  # For NVHPC we will rely on FE exceptions as opposed to errno in order to make use of faster
+  # builtins. One caveat is that if we use an optimization level greater than -O1, the FE exception
+  # is not raised. See https://github.com/neuronsimulator/nrn/pull/1930
+  set_source_files_properties(
+    ${PROJECT_SOURCE_DIR}/src/oc/math.cpp PROPERTIES COMPILE_DEFINITIONS NVHPC_CHECK_FE_EXCEPTIONS=1
+                                                     COMPILE_OPTIONS -O1)
+
```

Comment:
I suppose someday the -O1 will no longer be needed

Reply:
yeah...I hope this is "temporary" solution with NVHPC compiler.